src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/UrlRewriteFileParser.cs

```diff
@@ -138,61 +138,74 @@ private void ParseCondition(XElement condition, UrlRewriteRuleBuilder builder, P
+                        break;
+                    case ActionType.Rewrite:
+                    case ActionType.Redirect:
+                        var url = string.Empty;
+                        if (urlAction.Attribute(RewriteTags.Url) != null)
+                        {
+                            url = urlAction.Attribute(RewriteTags.Url).Value;
+                            if (string.IsNullOrEmpty(url))
+                            {
+                                throw new InvalidUrlRewriteFormatException(urlAction, "Url attribute cannot contain an empty string");
+                            }
+                        }
+
+                        var urlPattern = _inputParser.ParseInputString(url);
+                        var appendQuery = ParseBool(urlAction, RewriteTags.AppendQueryString, defaultValue: true);
+
+                        if (actionType == ActionType.Rewrite)
+                        {
+                            action = new RewriteAction(stopProcessing ? RuleResult.SkipRemainingRules : RuleResult.ContinueRules, urlPattern, appendQuery);
+                        }
+                        else
+                        {
+                            var redirectType = ParseEnum(urlAction, RewriteTags.RedirectType, RedirectType.Permanent);
+
+                            action = new RedirectAction((int)redirectType, urlPattern, appendQuery);
+                        }
+                        break;
+                    case ActionType.AbortRequest:
+                        action = new AbortAction();
+                        break;
+                    case ActionType.CustomResponse:
+                        int statusCode;
+                        if (!int.TryParse(urlAction.Attribute(RewriteTags.StatusCode)?.Value, out statusCode))
+                        {
+                            throw new InvalidUrlRewriteFormatException(urlAction, "A valid status code is required");
+                        }
+
+                        string statusReason = urlAction.Attribute(RewriteTags.StatusReason)?.Value;
+                        string statusDescription = urlAction.Attribute(RewriteTags.StatusDescription)?.Value;
+
+                        action = new CustomResponseAction(statusCode, statusReason, statusDescription);
+                        break;
+                    default:
+                        throw new NotSupportedException($"The action type {actionType} wasn't recognized");
                 }
+                builder.AddUrlAction(action);
             }
-
-            try
-            {
-                var input = _inputParser.ParseInputString(url);
-                builder.AddUrlAction(input, actionType, appendQuery, stopProcessing, (int)redirectType);
-            }
-            catch (FormatException formatException)
+            catch (Exception exception) when (!(exception is InvalidUrlRewriteFormatException))
```

Comment:
Don't do a catch-all-and-wrap like this. Are there any specific exceptions you're expecting?